sys/posix/netdb/netdb.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup posix
+ *
+ * @{
+ * @file
+ * @brief  posix netdb implementation
+ * @author Hendrik van Essen <hendrik.ve@fu-berlin.de>
+ * @}
+ */
+
+#include "netdb.h"
+
+struct hostent* gethostbyname(const char *name) {
+#if LWIP_NETDB
+    return lwip_gethostbyname(name);
+#else
+    #error "There is not implementation of gethostbyname available"
+#endif
```

Comment:
For all the functions here: I think it might be cleaner (and smaller) to provide our own implementation utilizing either `sock_dns` or `sock_udp` directly here. This would have multiple advantages:

- no stack-wise `#ifdefs`
- other DNS protocols than the standard DNS over UDP (see #16705 and #16861) and even completely different name-resolution protocols ([`netdb.h` spec does not mention DNS at all](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/basedefs/netdb.h.html) and [`getaddrinfo` spec just says "In many cases it is implemented by the Domain Name System, […]"](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/functions/getaddrinfo.html)) could be used here.

BTW, since I looked at the specs: `gethostbyname()` and its re-entrant counter-part where removed in POSIX.1-2008, so do we really want to support them and not rather fix libraries that still use them?